models/offer.go

```diff
@@ -222,11 +237,12 @@ func GetAvailableOffers(db runner.Connection, playerID, gameID string, t time.Ti
 			offerToReturn.RemainingPurchases = p.Max
 		}
 		o := &Offer{
-			GameID:          ot.GameID,
-			OfferTemplateID: ot.ID,
-			PlayerID:        playerID,
+			GameID:           ot.GameID,
+			OfferTemplateID:  ot.ID,
+			OfferTemplateKey: ot.Key,
+			PlayerID:         playerID,
 		}
-		playerOffer, playerHasOffer := playerOffersByOfferTemplateID[ot.ID]
+		playerOffer, playerHasOffer := playerOffersByOfferTemplateKey[ot.Key]
```

Comment:
Here the check should really be done with the offer template id. If the player has an offer that refers an older version of the template (same key, but different id) we'd like to create a new offer with the most recent info.